addon/-private/system/relationships/state/has-many.js

```diff
@@ -111,6 +111,32 @@ export default class ManyRelationship extends Relationship {
     super.removeCanonicalInternalModelFromOwn(internalModel, idx);
   }
 
+  unloadInverseInternalModelFromOwn(internalModel) {
+    super.unloadInverseInternalModelFromOwn(internalModel);
+
+    const canonicalIndex = this.canonicalState.indexOf(internalModel);
+
+    if (canonicalIndex !== -1) {
+      this.canonicalState.splice(canonicalIndex, 1);
+    }
+
+    const manyArray = this._manyArray;
+
+    if (manyArray) {
+      const currentState = manyArray.currentState;
+      const idx = currentState.indexOf(internalModel);
+
+      if (idx !== -1) {
+        currentState.splice(idx, 1);
+
+        manyArray.set('length', currentState.length);
+        manyArray.notifyPropertyChange('length');
+
+        this.internalModel.notifyHasManyRemoved(this.key, internalModel, idx);
```

Comment:
scratch that. 

The issue is, ManyArray emulates Enumerable, and thus must provide it's own `arrayContentWillChange` and `arrayContentDidChange` events. It's internal methods do this correctly, but your manual and external `currentState.splice` + `manyArray.set('length` does not. Rather then reaching into the internal state, you may want to use `manyArray.internalReplace` or add your own `manyArray.splice.

```js
splice(idx, count) {
    this.arrayContentWillChange(idx, count, 0);
    this.currentState.splice(idx, count);

    this.set('length', this.currentState.length);
    this.arrayContentDidChange(idx, count, 0);
  }
```